src/NServiceBus.Core/Transports/Receiving.cs

```diff
@@ -15,17 +15,24 @@ internal Receiving()
             Defaults(s =>
             {
                 var transportInfrastructure = s.Get<TransportInfrastructure>();
-                var userDiscriminator = s.GetOrDefault<string>("EndpointInstanceDiscriminator");
-                var baseQueueName = s.GetOrDefault<string>("BaseInputQueueName") ?? s.EndpointName();
-                if (userDiscriminator != null)
+                var discriminator = s.GetOrDefault<string>("EndpointInstanceDiscriminator");
+                var baseQueueName = s.GetOrDefault<string>("BaseInputQueueName");
+                var mainInstance = new EndpointInstance(s.EndpointName());
+                if (baseQueueName != null)
                 {
-                    var p = s.Get<TransportInfrastructure>().BindToLocalEndpoint(new EndpointInstance(baseQueueName, userDiscriminator));
-                    s.SetDefault("NServiceBus.EndpointSpecificQueue", transportInfrastructure.ToTransportAddress(new LogicalAddress(p)));
+                    mainInstance = mainInstance.SetProperty("queue", baseQueueName);
```

Comment:
@timbussmann how about setting the `queue` property **and** passing the overridden queue name as the endpoint name to the transport? Transports that are not aware of `queue` will work just fine without any modifications.